### PR TITLE
Complete Task3

### PR DESCRIPTION
**Pull request by NDragneelL9**
1. Add UserPersonality table
2. Add one to many association between user and user_personality
3. Create user_personality_controller to handle CRUD operations on user_personality model
4. Change home page
5. Create Views to CRUD operations on user_personality model
6. Restrict ability to delete last remaining user_personality for user
7. Add feature to create default user_personality when creating a user
8. Add rspec-rails, simplecov gems to cover writed code
9. Creat tests for models: user, user_personality and controller: user_personality_controller
Note:
User will see list of withdrawals and credits for each personality after choosing on '/user_personalities' path.
On root page user will see list of withdrawals and credits for default personality

### CODE REVIEW

File: app/models/user_personality.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class UserPersonality < ApplicationRecord
+  belongs_to :user
+  before_destroy :ensure_one_personality_remains
+  validates :name, presence: true,
+                   length: { minimum: 3, maximum: 14 },
+                   uniqueness: { case_sensitive: false, scope: :user_id,
+                                 message: 'for personality already exists for this user' }
+
+  def ensure_one_personality_remains
```
Comment (e-popolamov) [inline app/models/user_personality.rb:11]: make this method private

File: app/models/user_personality.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    throw(:abort) if user.user_personalities.count == 1
```
Comment (e-popolamov) [inline app/models/user_personality.rb]: better use raise ActiveRecord::Rollback

File: app/views/user_personalities/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div class="valign-wrapper row login-form">
+  <div class="col card hoverable s10 pull-s1 m6 pull-m3 l4 pull-l4">
+    <div class="card-content">
+      <span class="card-title center"><%= @personality.id.nil? ? 'Create personality' : 'Update personality' %></span>
+      <div class="divider"></div>
+      <%= render 'shared/error_messages', obj: @personality %>
+      <%= form_with(model: @personality, local: true) do |f| %>
+        <div class="form-inputs row">
+          <div class="input-field col s12">
+            <%= f.text_field :name, required: true, autofocus: true, placeholder: "Wife" %>
```
Comment (e-popolamov) [inline app/views/user_personalities/_form.html.erb:10]: why "wife"?

**Comment by e-popolamov** [comment]
Also i found bug when user try to get user_personalities with not existing id (for ex. /user_personalities/0)

**Review by e-popolamov — CHANGES_REQUESTED** [review]
Good coverage by rspec and rubocop.
Please add fix and request me again

File: app/views/user_personalities/_form.html.erb (hunk 3 above)
Comment (NDragneelL9) [reply app/views/user_personalities/_form.html.erb:10]: cuz its first thing that comes to my head when some male talks about withdrawals

**Review by e-popolamov — APPROVED** [review]
Good job!

**Pull request merged**